system/jlib/jthread.cpp

```diff
@@ -2162,9 +2166,8 @@ protected: friend class PipeWriterThread;
                     throw MakeStringException(-1, "CLinuxPipeProcess::run: could not change dir to %s", dir.get());
```

Comment:
Unrelated to this change, but the comment on lines 2139 suggests (for several reasons) this code is not safe, also code in 2173->

Reply:
Both are failure conditions that SHOULD be terminating the child process. I can raise a separate Jira to address them, but I suspect it will just be a matter of putting an error out to stderr and calling _exit